src/main/java/Lemon.java

```diff
@@ -0,0 +1,286 @@
+                        addNewTask(new Todo(temp[2], Boolean.parseBoolean(temp[1])));
+                        break;
+                    case "D":
+                        addNewTask(new Deadline(temp[2],
+                                LocalDate.parse(temp[3], DateTimeFormatter.ofPattern("yyyy-MM-dd")),
+                                Boolean.parseBoolean(temp[1])));
+                        break;
+                    case "E":
+                        addNewTask(new Event(temp[2],
+                                LocalDate.parse(temp[3], DateTimeFormatter.ofPattern("yyyy-MM-dd")),
+                                LocalDate.parse(temp[4], DateTimeFormatter.ofPattern("yyyy-MM-dd")),
+                                Boolean.parseBoolean(temp[1])));
+                        break;
+                }
+            }
+
+            isInitialised = true;
+        } catch (IOException e) {
+            isInitialised = false;
+            System.out.print(" Im sowwy... Something went wrong, QwQ. Unable to create file.\n" +
+                    " I dont think i can do this anymore");
+        } catch (DescriptionException e) {
+            isInitialised = false;
+            System.out.println(e.getMessage());
+            // theres an invalid line
+        }
+    }
+
+    public Lemon(String newFileLocation){
+        try {
+            filePath = newFileLocation;
+            f = new File(filePath);
+            f.getParentFile().mkdirs();
+            f.createNewFile();
+            fileScanner = new Scanner(f);
+            systemScanner = new Scanner(System.in);
+
+            numTasks = 0;
+            tasks = new ArrayList<>();
+            while (fileScanner.hasNextLine()) {
+                numTasks++;
+                fileScanner.nextLine();
+            }
+
+            isInitialised = true;
+        } catch (IOException e) {
+            isInitialised = false;
+            System.out.print(" Im sowwy... Something went wrong, QwQ. Unable to create/find file.\n" +
+                    " I dont think i can do this anymore");
+        }
+    }
+
+    public void runLemon() {
+        if (!isInitialised) {
+            System.out.print(" Ouhiiee, my head hurrrtt, i dont think im initialized properly ;-;\n" +
+                    " Please try again...");
+            return;
+        }
+
+        boolean isRunning = true;
+
+        System.out.print(logoMsg);
+
+        while (isRunning) {
+            System.out.println(barMsg);
+
+            String input = systemScanner.next().toUpperCase();
+            System.out.println(barMsg);
+            try {
+                Commands command = Commands.valueOf(input);
+                switch (command) {
+                case BYE: {
+                    boolean isAbleToSave = saveTasks();
+                    if (isAbleToSave) {
+                        isRunning = false;
+                        System.out.println(endMsg);
+                    }
+                    break;
+                }
```

Comment:
Correct indentation for case in this switch statement. 

Though, braces aren't required. It can just be:
```java
switch (command) {
case BYE:
    boolean isAbleToSave = ...
    // ...
    break;
```